raster/r.info/main.c

```diff
@@ -440,81 +461,149 @@ int main(int argc, char **argv)
-            fprintf(out, "ewres=%s\n", tmp3);
-
-            fprintf(out, "rows=%d\n", cellhd.rows);
-            fprintf(out, "cols=%d\n", cellhd.cols);
-
-            fprintf(out, "cells=%" PRId64 "\n",
-                    (grass_int64)cellhd.rows * cellhd.cols);
-
-            fprintf(out, "datatype=%s\n",
-                    (data_type == CELL_TYPE
-                         ? "CELL"
-                         : (data_type == DCELL_TYPE
-                                ? "DCELL"
-                                : (data_type == FCELL_TYPE ? "FCELL" : "??"))));
-            if (cats_ok)
-                format_double((double)cats.num, tmp1);
-            fprintf(out, "ncats=%s\n", cats_ok ? tmp1 : "??");
+        if (gflag->answer || format == JSON) {
+            const char *data_type_f =
+                (data_type == CELL_TYPE
+                     ? "CELL"
+                     : (data_type == DCELL_TYPE
+                            ? "DCELL"
+                            : (data_type == FCELL_TYPE ? "FCELL" : "??")));
+            grass_int64 total_cells = (grass_int64)cellhd.rows * cellhd.cols;
+
+            switch (format) {
+            case PLAIN:
+                G_format_northing(cellhd.north, tmp1, -1);
+                G_format_northing(cellhd.south, tmp2, -1);
+                fprintf(out, "north=%s\n", tmp1);
+                fprintf(out, "south=%s\n", tmp2);
+
+                G_format_easting(cellhd.east, tmp1, -1);
+                G_format_easting(cellhd.west, tmp2, -1);
+                fprintf(out, "east=%s\n", tmp1);
+                fprintf(out, "west=%s\n", tmp2);
+
+                G_format_resolution(cellhd.ns_res, tmp3, -1);
+                fprintf(out, "nsres=%s\n", tmp3);
+
+                G_format_resolution(cellhd.ew_res, tmp3, -1);
+                fprintf(out, "ewres=%s\n", tmp3);
+
+                fprintf(out, "rows=%d\n", cellhd.rows);
+                fprintf(out, "cols=%d\n", cellhd.cols);
+
+                fprintf(out, "cells=%" PRId64 "\n", total_cells);
+
+                fprintf(out, "datatype=%s\n", data_type_f);
+
+                if (cats_ok)
+                    format_double((double)cats.num, tmp4);
+                fprintf(out, "ncats=%s\n", cats_ok ? tmp4 : "??");
+                break;
+            case JSON:
+                json_object_set_number(root_object, "north", cellhd.north);
+                json_object_set_number(root_object, "south", cellhd.south);
+                json_object_set_number(root_object, "nsres", cellhd.ns_res);
+
+                json_object_set_number(root_object, "east", cellhd.east);
+                json_object_set_number(root_object, "west", cellhd.west);
+                json_object_set_number(root_object, "ewres", cellhd.ew_res);
+
+                json_object_set_number(root_object, "rows", cellhd.rows);
+                json_object_set_number(root_object, "cols", cellhd.cols);
+                json_object_set_number(root_object, "cells", total_cells);
```

Comment:
Return an integer for the value of rows,, cols, and cells.

Reply:
Fixed by changing to default format specifier for parson.